presto-iceberg/src/test/java/com/facebook/presto/iceberg/nessie/TestIcebergSmokeNessie.java

```diff
@@ -111,4 +122,68 @@ protected Table getIcebergTable(ConnectorSession session, String schema, String
                 session,
                 SchemaTableName.valueOf(schema + "." + tableName));
     }
+
+    @DataProvider
+    public Object[][] concurrencyValues()
+    {
+        return new Object[][] {
+                {4},
+                {5}
+        };
+    }
+
+    @Test(dataProvider = "concurrencyValues")
+    public void testSuccessfulConcurrentInsert(int concurrency)
+    {
+        final Session session = getSession();
+        assertUpdate(session, "CREATE TABLE test_concurrent_insert (col0 INTEGER, col1 VARCHAR) WITH (format = 'ORC', commit_retries = " + concurrency + ")");
+
+        int commitRetries = concurrency + 1;
+        final String[] strings = {"one", "two", "three", "four", "five", "six", "seven"};
+        final CountDownLatch countDownLatch = new CountDownLatch(commitRetries);
+        AtomicInteger value = new AtomicInteger(0);
+        Set<Throwable> errors = new CopyOnWriteArraySet<>();
+        List<Thread> threads = Stream.generate(() -> new Thread(() -> {
```

Comment:
There are too many tests that try to accomplish similar tasks through multi-threading.  Ultimately, these style of tests worry me because they end up being highly environment-specific--suspectible to GC pauses, noisy neighbors and such.  Can we think of a different way to accomplish this that isn't time depenendent?

Reply:
I am currently working on this. Lets see if I am able to come up with something.

Reply:
I would recommend just removing the test, since this is a table property and a functionality of the core Iceberg library.  Adding a potentially flaky test to our suites is just not worth the potential problems.

Reply:
@tdcmeehan So with the current approach of submitting 100 queries, the test case is able to pass consistently. I am now only trying to fine tune the number of threads and other parameters, so as to reduce the test run time and not introduce flakiness overall. 
Let me know if you still think otherwise. We can probably discuss internally on this. 
cc @agrawalreetika @imjalpreet @yingsu00 @nmahadevuni